lib/location/location_core.c

```diff
@@ -617,6 +620,42 @@ static void location_core_event_details_get(struct location_event_data *event)
 #endif
 }
 
+static void location_core_event_cb_fallback(
+	enum location_method current_method,
+	enum location_method next_method,
+	enum location_event_id current_event,
+	struct location_data_details *details)
+{
+#if defined(CONFIG_LOCATION_DATA_DETAILS)
+	if (current_method == LOCATION_METHOD_INTERNAL_WIFI_CELLULAR) {
+		/* When fallback happens, current method is always the 1st one whenever
```

Comment:
As already discussed, I had hard time understanding why this works. 😄 It could be only me, but you could consider explaining in even more detail why indexes 0 and 1 can be used here.

And maybe we should consider limiting the maximum value of `CONFIG_LOCATION_METHODS_LIST_SIZE` to 3. There should be no reason to increase it past 3, but increasing it could break this implementation.